lib/room.cpp

```diff
@@ -354,37 +354,95 @@ class Room::Private {
     // A map from senderKey to a map of sessionId to InboundGroupSession
     // Not using QMultiHash, because we want to quickly return
     // a number of relations for a given event without enumerating them.
-    QHash<QPair<QString, QString>, QOlmInboundGroupSession*> groupSessions; // TODO:
-                                                                        // cache
+    std::map<QPair<QString, QString>, std::unique_ptr<QOlmInboundGroupSession>> groupSessions;
+
+    void loadMegOlmSessions() {
```

Comment:
I'm understanding it correctly that this will load every session for each room and do that for each room? Shouldn't we move that to the encryptionmanager?

Reply:
Ah, right, i forgot to make those specific to the rooms